### ISSUE TITLE
fatal when set lr_mult=0

### ISSUE THREAD

**Issue by tangjiuqi097** [sfzhang15/RefineDet#109]
I add additional layers and train with the original layers' parameters fixed, so I set all the original layers' lr_mult equal to 0. But it raises error. 
I try to remove the additional layers to make sure the problem is not caused by additional layers, the error still exist.

The message is 

I0816 20:20:26.116406 25974 caffe.cpp:251] Starting Optimization
I0816 20:20:26.116479 25974 solver.cpp:294] Solving coco_refinedet_vgg16_320x320_train
I0816 20:20:26.116487 25974 solver.cpp:295] Learning Rate Policy: multistep
I0816 20:20:26.457132 25974 solver.cpp:243] Iteration 0, loss = 6.09376
I0816 20:20:26.457157 25974 solver.cpp:259]     Train net output #0: arm_loss = 3.4709 (* 1 = 3.4709 loss)
I0816 20:20:26.457162 25974 solver.cpp:259]     Train net output #1: odm_loss = 2.62287 (* 1 = 2.62287 loss)
I0816 20:20:26.457173 25974 sgd_solver.cpp:138] Iteration 0, lr = 0.001
F0816 20:20:26.625644 25974 bbox_util.cu:779] Check failed: match_index[p] < gt_bboxes.size() (16 vs. 1) 
*** Check failure stack trace: ***
    @     0x7f6dfe0a95cd  google::LogMessage::Fail()
    @     0x7f6dfe0ab433  google::LogMessage::SendToLog()
    @     0x7f6dfe0a915b  google::LogMessage::Flush()
    @     0x7f6dfe0abe1e  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f6dfea38ecd  caffe::ComputeConfLossGPU<>()
    @     0x7f6dfe9aa58e  caffe::MineHardExamples<>()
    @     0x7f6dfe79e4ec  caffe::MultiBoxLossLayer<>::Forward_cpu()
    @     0x7f6dfe776b87  caffe::Net<>::ForwardFromTo()
    @     0x7f6dfe776ef7  caffe::Net<>::Forward()
    @     0x7f6dfe9c2a20  caffe::Solver<>::Step()
    @     0x7f6dfe9c34ae  caffe::Solver<>::Solve()
    @           0x40bd14  train()
    @           0x4077e8  main
    @     0x7f6dfc83f830  __libc_start_main
    @           0x4080b9  _start
    @              (nil)  (unknown)
Aborted (core dumped)

**Comment by sfzhang15** [comment]
@tangjiuqi097 
According to the error message (Check failed: match_index[p] < gt_bboxes.size() (16 vs. 1)), the error is about GT, not setting lr_mult=0. If you unchange the lr_mult, the error should also exist, right?

**Comment by sfzhang15** [comment]
@tangjiuqi097 
I have changed all the lr_mult to 0 and it runs successfully.
![image](https://user-images.githubusercontent.com/31090111/44445603-309dba00-a614-11e8-8633-0288b054047b.png)
![image](https://user-images.githubusercontent.com/31090111/44445620-3eebd600-a614-11e8-955d-c9b6e373330d.png)

**Comment by tangjiuqi097** [comment]
@sfzhang15 
Do you set **all** the lr_mult to zero? 

I notice that set the lr_mult=0 in RefineDet/models/VGGNet/coco/refinedet_vgg16_320x320/VGG16_COCO_320.py will not change all the lr_mult.

And two Normalize layers does not set the lr_mult, so it use the default lr_mult=1. It needs to add param to set the lr_mult manually in RefineDet/models/VGGNet/coco/refinedet_vgg16_320x320/train.prototxt.

Could you try again?

The RefineDet/models/VGGNet/coco/refinedet_vgg16_320x320/train.prototxt that I set all lr_mult=0 is
[set_all_lr_mult_0_train_prototxt.txt](https://github.com/sfzhang15/RefineDet/files/2312709/set_all_lr_mult_0_train_prototxt.txt)

**Comment by tangjiuqi097** [comment]
@sfzhang15 
Thanks. If you have any idea about it, please remind me.
